### DISCUSSION TITLE
How to validate a list of django model instances.

### DISCUSSION THREAD

**Discussion by lubiah** [pydantic/pydantic#11185] [Question] [answered]
Hello,
I'm having a django user model which I've been able to create a pydantic serializer for.
```python
class User(BaseModel):
   first_name: str
   last_name: str

```
I've used a TypedAdapter to create a serializer for a list of users like UsersSerializer = TypedAdapter(list[User]).
Since the user instances are coming from an ORM, how do I validate it??
Should I use dump_python first and then pass the results to validate python???

**Comment by Viicos** [comment, marked as answer]
You can read about the `from_attributes` configuration value here: https://docs.pydantic.dev/dev/concepts/models/#arbitrary-class-instances.

Alternatively, some libraries like [SQLModel](https://github.com/fastapi/sqlmodel/) can provide integrations to have both the Pydantic model and ORM class (in the case of SQLModel, SQLAlchemy) as a single type, but this is imo a slippery slope.

Usually, it's best to have the two decoupled, for instance, this is [the approach taken by the Litestar](https://docs.litestar.dev/latest/usage/dto/index.html) framework.
